optee-utee/src/extension.rs

```diff
@@ -26,32 +26,304 @@ pub struct LoadablePlugin {
     uuid: Uuid
 }
 
+pub struct LoadablePluginCommand<'a> {
+    plugin: &'a LoadablePlugin,
+    cmd_id: u32,
+    sub_cmd_id: u32,
+    buffer: Vec<u8>,
+}
+
 impl LoadablePlugin {
     pub fn new(uuid: &Uuid) -> Self {
         Self { uuid: uuid.to_owned() }
     }
-    pub fn invoke(&mut self, command_id: u32, subcommand_id: u32, data: &[u8]) -> Result<Vec<u8>> {
-        let raw_uuid: Uuid = self.uuid;
+    pub fn invoke(&self, command_id: u32, subcommand_id: u32, data: &[u8]) -> Result<Vec<u8>> {
```

Comment:
Can we have some comments on each functions to provide guidance for developers, such as the suitable scenario for `invoke()` and `invoke_with_capacity()`? The comments can also be rendered into our API documentation. 
FYI One example for the comment format: https://github.com/apache/incubator-teaclave-trustzone-sdk/blob/20e698863932d395b48002e5fb126491e0caef0d/optee-utee/src/object.rs#L83-L94